Chat/src/app/ChatScreen.tsx

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { CommunicationUserIdentifier } from '@azure/communication-common';
+import {
+  AvatarPersonaData,
+  ChatAdapter,
+  ChatComposite,
+  createAzureCommunicationChatAdapter,
+  fromFlatCommunicationIdentifier
+} from '@azure/communication-react';
+import { Stack } from '@fluentui/react';
+import React, { useEffect, useRef, useState } from 'react';
+
+import { ChatHeader } from './ChatHeader';
+import { chatCompositeContainerStyle, chatScreenContainerStyle } from './styles/ChatScreen.styles';
+import { createAutoRefreshingCredential } from './utils/credential';
+import { fetchEmojiForUser } from './utils/emojiCache';
+import { getBackgroundColor } from './utils/utils';
+import { useSwitchableFluentTheme } from './theming/SwitchableFluentThemeProvider';
+
+// These props are passed in when this component is referenced in JSX and not found in context
+interface ChatScreenProps {
+  token: string;
+  userId: string;
+  displayName: string;
+  endpointUrl: string;
+  threadId: string;
+  endChatHandler(isParticipantRemoved: boolean): void;
+  errorHandler(): void;
+}
+
+export const ChatScreen = (props: ChatScreenProps): JSX.Element => {
+  const { displayName, endpointUrl, threadId, token, userId, errorHandler, endChatHandler } = props;
+
+  const adapterRef = useRef<ChatAdapter>();
+  const [adapter, setAdapter] = useState<ChatAdapter>();
+  const [hideParticipants, setHideParticipants] = useState<boolean>(false);
+  const { currentTheme } = useSwitchableFluentTheme();
+
+  useEffect(() => {
+    (async () => {
+      const adapter = await createAzureCommunicationChatAdapter({
+        endpoint: endpointUrl,
+        userId: fromFlatCommunicationIdentifier(userId) as CommunicationUserIdentifier,
+        displayName: displayName,
+        credential: createAutoRefreshingCredential(userId, token),
+        threadId: threadId
+      });
+      adapter.on('participantsRemoved', (listener) => {
+        // Note: We are receiving ChatParticipant.id from communication-signaling, so of type 'CommunicationIdentifierKind'
+        // while it's supposed to be of type 'CommunicationIdentifier' as defined in communication-chat
+        const removedParticipantIds = listener.participantsRemoved.map(
+          (p) => (p.id as CommunicationUserIdentifier).communicationUserId
+        );
+        if (removedParticipantIds.includes(userId)) {
+          const removedBy = (listener.removedBy.id as CommunicationUserIdentifier).communicationUserId;
+          endChatHandler(removedBy !== userId);
+        }
+      });
+      adapter.setTopic('Your Chat sample');
```

Comment:
Does this mean anytime someone is removed it resets the topic to Your Chat sample if it had been set to something new?

Reply:
Created an ADO item
https://skype.visualstudio.com/SPOOL/_workitems/edit/2691389